File: src/CitelTestPratico.Dominio/ProdutoRoot/Produto.cs
using CitelTestPratico.Dominio.CategoriaProdutoRoot;
using CitelTestPratico.Dominio.Core.Models;

namespace CitelTestPratico.Dominio.ProdutoRoot
{
    public class Produto : Entity
    {
        public string Nome { get; set; }
        public string CodigoBarra { get; set; }
        public string Descricao { get; set; }
        public bool Ativo { get; set; }
        public string UriImageDefault { get; set; }
        public List<CategoriaProduto> Categorias { get; set; }
    }
}

File: src/CitelTestPratico-Fornt-End/src/app/pages/produto/models/produto.ts
import { Categoria } from "../../categoria/models/categoria";

export class Produto {
    id!: string;
    nome!: string;
    codigoBarra!: string;
    ativo!: boolean;
    descricao!: string;
    uriImageDefault!: string;
    categoriasIds!: number[];
    categorias!: Categoria[];
}

